composites/AlgoliaSearch/AlgoliaSearcher.js

```diff
@@ -141,182 +165,146 @@ class AlgoliaSearcher extends React.Component {
 	}
 
 	/**
-	 * Sets all values required to display the detail view of a search result.
+	 * Adds the data from the clicked result to the information associated with the current search string.
 	 *
-	 * @param {number} resultIndex The index of the article you want to
-	 *                                  show in the state.results array.
-	 * @returns {void}
+	 * @param {number} currentDetailViewIndex The current detail view index.
+	 *
+	 * @returns {Object} Object containing the currently used queries with additional post data.
 	 */
-	showDetail( resultIndex ) {
+	addResultToUsedQueries( currentDetailViewIndex ) {
+		let post = this.state.results[ currentDetailViewIndex ];
 		let usedQueries = this.state.usedQueries;
-		let post = this.state.results[ resultIndex ];
 
 		usedQueries[ this.state.searchString ][ post.objectID ] = {
-			title: post.post_title,
+			title: post.postTitle,
 			link: post.permalink,
 		};
 
-		this.setState( {
-			showDetail: resultIndex,
-			usedQueries: usedQueries,
-		} );
+		return usedQueries;
 	}
 
 	/**
-	 * Hides the details page and return to the results page.
+	 * Sets all values required to display the detailed view of a search result.
+	 *
+	 * @param {number} currentDetailViewIndex The index of the article you want to show in the state.results array.
 	 *
 	 * @returns {void}
 	 */
-	hideDetail() {
-		this.setState( { showDetail: false } );
+	setCurrentDetailedViewIndex( currentDetailViewIndex ) {
+		let usedQueries = this.addResultToUsedQueries( currentDetailViewIndex );
+
+		this.setState( { currentDetailViewIndex, usedQueries } );
 	}
 
 	/**
-	 * Renders a no results found text.
+	 * Resets the values associated with the displaying of the detailed search result view.
 	 *
-	 * @returns {ReactElement} The no results found text.
+	 * @returns {void}
 	 */
-	renderNoResultsFound() {
-		let searchResultContent = <p>{ this.props.noResultsText }</p>;
-		a11ySpeak( this.props.noResultsText );
-
-		return searchResultContent;
+	unsetCurrentDetailedView() {
+		this.setState( { currentDetailViewIndex: -1 } );
 	}
 
 	/**
-	 * Maps the results to SearchResult components.
+	 * Log any occuring error and render a search error warning.
 	 *
-	 * @param {object} results The results returned by Algolia.
+	 * @param {string} errorMessage The message to display.
 	 *
-	 * @returns {Array} Array containing the mapped search results.
+	 * @returns {ReactElement} A p tag with a warning that the search was not completed.
```

Comment:
Technically, a "tag" is what you use to in the markup language. This return a p element instead.